test/immediate_sync_test.cpp

```diff
@@ -88,3 +88,90 @@ TEST_F(ManagerTest, testDataChangeInFile)
 
     ctx.run();
 }
+
+TEST_F(ManagerTest, testDataChangeDisableProperty)
+{
+    using namespace std::literals;
+    namespace extData = data_sync::ext_data;
+
+    std::unique_ptr<extData::ExternalDataIFaces> extDataIface =
+        std::make_unique<extData::MockExternalDataIFaces>();
+
+    extData::MockExternalDataIFaces* mockExtDataIfaces =
+        dynamic_cast<extData::MockExternalDataIFaces*>(extDataIface.get());
+
+    ON_CALL(*mockExtDataIfaces, fetchBMCRedundancyMgrProps())
+        // NOLINTNEXTLINE
+        .WillByDefault([&mockExtDataIfaces]() -> sdbusplus::async::task<> {
+        mockExtDataIfaces->setBMCRole(extData::BMCRole::Active);
+        co_return;
+    });
+
+    EXPECT_CALL(*mockExtDataIfaces, fetchSiblingBmcIP())
+        // NOLINTNEXTLINE
+        .WillRepeatedly([]() -> sdbusplus::async::task<> { co_return; });
+
+    EXPECT_CALL(*mockExtDataIfaces, fetchRbmcCredentials())
+        // NOLINTNEXTLINE
+        .WillRepeatedly([]() -> sdbusplus::async::task<> { co_return; });
+
+    nlohmann::json jsonData = {
+        {"Files",
+         {{{"Path", ManagerTest::tmpDataSyncDataDir.string() + "/srcFile2"},
+           {"DestinationPath",
+            ManagerTest::tmpDataSyncDataDir.string() + "/destFile2"},
+           {"Description", "File to test immediate sync upon data write"},
+           {"SyncDirection", "Active2Passive"},
+           {"SyncType", "Immediate"}}}}};
+
+    std::string srcPath{jsonData["Files"][0]["Path"]};
+    std::string destPath{jsonData["Files"][0]["DestinationPath"]};
+
+    writeConfig(jsonData);
+    sdbusplus::async::context ctx;
+
+    std::string data{"Src: Initial Data\n"};
+    // Just for create path in the dest to watch for expectation check
+    std::string destData{"Dest: Initial Data\n"};
+    ManagerTest::writeData(srcPath, data);
+    ManagerTest::writeData(destPath, destData);
+    ASSERT_EQ(ManagerTest::readData(srcPath), data);
+    ASSERT_EQ(ManagerTest::readData(destPath), destData);
+
+    data_sync::Manager manager{ctx, std::move(extDataIface),
+                               ManagerTest::dataSyncCfgDir};
+    manager.setDisableSyncStatus(true); // disabled the sync events
+
+    EXPECT_NE(ManagerTest::readData(destPath), data)
+        << "The data should not match because the manager is spawned and"
+        << " sync is disabled.";
+
+    std::string dataToWrite{"Data is modified"};
+    std::string dataToStopEvent{"Close spawned inotify event."};
+
+    // write data to src path to create inotify event so that the spawned
+    // process is closed.
+    ManagerTest::writeData(srcPath, dataToStopEvent);
```

Comment:
is it guarantee that background sync will be ready to catch this? I think we added some delay in the previous commits to guarantee that.